File: app/src/main/java/com/bracketcove/postrainer/reminderdetail/ReminderDetailContract.java
package com.bracketcove.postrainer.reminderdetail;

import com.bracketcove.postrainer.base.BasePresenter;
import com.bracketcove.postrainer.base.BaseView;
import com.bracketcove.postrainer.data.viewmodel.Reminder;

/**
 * Created by Ryan on 06/03/2017.
 */

public interface ReminderDetailContract {
    interface View extends BaseView<Presenter> {
        Reminder getViewModel();

        void setReminderTitle(String title);

        void setVibrateOnly(boolean active);

        void setRenewAutomatically(boolean active);

        void setPickerTime(int hour, int minute);

        void setCurrentAlarmState(boolean active);

        String getReminderId();

        void startReminderListActivity();

    }

    interface Presenter extends BasePresenter {
        void onBackIconPress();

        void onDoneIconPress();
    }
}

File: app/src/main/java/com/bracketcove/postrainer/reminder/ReminderDetailContract.kt
package com.bracketcove.postrainer.reminder

import com.wiseassblog.domain.domainmodel.Reminder

/**
 * Created by Ryan on 06/03/2017.
 */

interface ReminderDetailContract {
    interface View  {
        fun setReminderTitle(title:String)

        fun setVibrateOnly(vibrateOnly: Boolean)

        fun setRenewAutomatically(renewAutomatically: Boolean)

        fun setPickerTime(hour: Int, minute: Int)

        fun startReminderListActivity()

        fun showMessage(msg: String)

        fun showDeleteConfirm()

        fun getState(): Reminder
    }

    interface ViewModel {
        fun getReminder(): Reminder?
        fun setReminder(reminder: Reminder)
    }
}

sealed class ReminderDetailEvent {
    object OnBackIconPress : ReminderDetailEvent()
    object OnDoneIconPress : ReminderDetailEvent()
    object OnDeleteIconPress : ReminderDetailEvent()
    object OnDeleteConfirmed : ReminderDetailEvent()
    object OnStart : ReminderDetailEvent()
}
